tests/test_submit.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
+
+import pathlib
+from typing import TYPE_CHECKING
+
+import pytest
+from pydantic import BaseModel
+
+if TYPE_CHECKING:
+    from pytest_mock import MockerFixture
+
+class CheckoutGitTestScenario(BaseModel):
+    debug_scenario_name: str
+    git_status_mocked_output: str = ""
+    internet_permissions: list[str] = []
+    git_push_result: tuple[int, str] | None = None # (exit_code, output_message)
+    expected_prompts: list[str] = []
+
+@pytest.fixture(autouse=True)
+def clear_modules():
```

Comment:
I like to prefix fixtures with `fixture_`: [reason](https://github.com/METR/mp4-tasks/pull/617#discussion_r1801215483)